public/apps/configuration/panels/audit-logging/code-editor.tsx

```diff
@@ -0,0 +1,72 @@
+/*
+ *   Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ *   Licensed under the Apache License, Version 2.0 (the "License").
+ *   You may not use this file except in compliance with the License.
+ *   A copy of the License is located at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *   or in the "license" file accompanying this file. This file is distributed
+ *   on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ *   express or implied. See the License for the specific language governing
+ *   permissions and limitations under the License.
+ */
+
+import React, { useState } from 'react';
+import { EuiCodeEditor, EuiText, EuiTextColor } from '@elastic/eui';
+import { get } from 'lodash';
+
+import 'brace/theme/textmate';
+import 'brace/mode/json';
+
+export function CodeEditor(props: {
+  setting: object;
+  config: object;
+  handleChange: () => void;
+  handleInvalid: () => void;
+}) {
+  const setting = props.setting;
+  const [value, updateValue] = useState(JSON.stringify(get(props.config, setting.path), null, 2));
+  const [invalid, setInvalid] = useState(false);
```

Comment:
Please add the type.

Reply:
This whole file will be removed and added later. I will address it when it is back.